compiler/rustc_driver_impl/src/lib.rs

```diff
@@ -370,6 +366,10 @@ pub fn run_compiler(at_args: &[String], callbacks: &mut (dyn Callbacks + Send))
 
             tcx.ensure_ok().analysis(());
 
+            if let Some(metrics_dir) = &sess.opts.unstable_opts.metrics_dir {
```

Comment:
Maybe ad a comment that no metrics may be written before `analysis` to avoid ICE.

Reply:
I am not confident that we will always use the same naming scheme involving the SVH for metrics going forward so I would prefer to put the comment on the `crate_hash` query or next to my call to `crate_hash`